### DISCUSSION TITLE
Option for `git checkout --merge`?

### DISCUSSION THREAD

**Discussion by gustav-b** [magit/magit#4889] [Support] [answered]
Is it possible to do the equivalent of `checkout`/`switch` with the `--merge` option from within Magit?

I sometimes have the situation where I've made uncommitted changes on the wrong branch and would like to switch to the right branch. If the uncommitted changes affect files that have other changes on the target branch, Git/Magit will refuse the switch with:
```
error: Your local changes to the following files would be overwritten by checkout:
	[…]
Please commit your changes or stash them before you switch branches.
Aborting
```
…so I have to stash, switch, unstash even though the changes are easily mergeable.

With `git checkout --merge` I could do this in one step, but I can't find an argument to Magit's branch checkout (`b b`) to do this. Am I missing something?

**Comment by kyleam** [comment, marked as answer]
Not sure if this is a common enough need across users to warrant adding to the magit-branch transient (perhaps at a hidden level), but note that you can add the switch yourself:

```elisp
(transient-append-suffix 'magit-branch "-r"
  '("-m" "Merge local modifications" "--merge"))
```

**Reply by gustav-b** [reply]
Ah, I wasn't aware it could be added like that. Thanks, that works!

**Reply by tarsius** [reply]
IMO it would make sense to add this but on a hidden level.

But can we come up with a better short description of what this does than "merge local modifications". Maybe "bring along uncommitted changes"? The fact that this might involve a merge is an unfortunate complication and the hope usually is that this won't happen. "merge" still appears in the name of the option, so we don't hide the potential complication.

**Reply by kyleam** [reply]
> Maybe "bring along uncommitted changes"?

Hmm, I find that confusing since that's the standard behavior without `--merge`.  It's just that, without `--merge`, `git checkout` is unwilling to continue if a file with uncommitted changes differs between the current branch and the checkout target.

I don't have any objections to changing the description I used above, but I haven't come up with a better one yet, aside from perhaps the minor tweak of "Merge local modifications if needed".

**Reply by tarsius** [reply]
Nevermind.

**Reply by baltakatei** [reply]
For people finding this answer via search engine, if you add the `(transient-append-suffix)` statement to your init file (e.g.  `~/emacs.d/init.el`), then I would also suggest wrapping it in `(with-eval-after-load)` as shown below so Emacs doesnʼt throw an error immediately upon startup, having not loaded magit stuff yet.

```
(with-eval-after-load 'magit
  (transient-append-suffix 'magit-branch "-r"
    '("-m" "Merge local modifications" "--merge")))
```
